vllm/worker/embedding_model_runner.py

```diff
@@ -137,14 +149,34 @@ def prepare_model_input(
         seq_group_metadata_list: Optional[List[SequenceGroupMetadata]],
         virtual_engine: int = 0,
         finished_requests_ids: Optional[List[str]] = None
-    ) -> ModelInputForGPUWithPoolingMetadata:
+    ) -> EmbeddingModelInput:
         assert seq_group_metadata_list is not None
         model_input = self._prepare_model_input_tensors(
             seq_group_metadata_list, finished_requests_ids)
+
+        (
+            attn_metadata,
+            encoder_input_tokens_tensor,
+            encoder_input_positions_tensor,
+            encoder_seq_lens,
+        ) = super()._prepare_encoder_model_input_tensors(
+            seq_group_metadata_list, model_input)
+
+        model_input = dataclasses.replace(
+            model_input,
+            attn_metadata=attn_metadata,
+            encoder_input_tokens=encoder_input_tokens_tensor,
+            encoder_input_positions=encoder_input_positions_tensor,
+        )
+
         # Prepare PoolingMetadata.
-        assert model_input.seq_lens is not None
+        seq_lens = model_input.seq_lens\
+            if not self.model_config.is_encoder_model \
```

Comment:
@robertgshaw2-neuralmagic @mgoin Here still have a question. How to determine precisely which model is encoder model or decoder model. As an example, there is lake of fields `is_decoder_model` or `is_encoder_model` in `config.json` of `Mistral Model` which is implemented before. Here is the link: [config.json of Mistral](https://huggingface.co/intfloat/e5-mistral-7b-instruct/blob/main/config.json).